qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,373 @@
+        """
+        if value <= 0:
+            raise ValueError("The size of the generation should be greater than 0.")
+        self._size_gen = value
+
+    @property
+    def max_iter(self) -> int:
+        """Returns the maximum number of iterations"""
+        return self._max_iter
+
+    @max_iter.setter
+    def max_iter(self, value: int):
+        """
+        Sets the maximum number of iterations of the algorithm.
+
+        Args:
+            value: Maximum number of iterations of the algorithm.
+
+        Raises:
+            ValueError: If `value` is lower than 1.
+        """
+        if value <= 0:
+            raise ValueError("The maximum number of iterations should be greater than 0.")
+
+        self._max_iter = value
+
+    @property
+    def alpha(self) -> float:
+        """Returns the alpha parameter value (percentage of population selected to update
+        probabilistic model)"""
+        return self._alpha
+
+    @alpha.setter
+    def alpha(self, value: float):
+        """
+        Sets the alpha parameter (percentage of individuals selected to update the probabilistic
+        model)
+
+        Args:
+            value: Percentage [0,1] of generation selected to update the probabilistic model.
+
+        Raises:
+            ValueError: If `value` is lower than 0 or greater than 1.
+        """
+        if (value < 0) or (value > 1):
+            raise ValueError(
+                "Alpha represents a percentage and should be greater than 0 but lower " "than 1"
+            )
+
+        self._alpha = value
+
+    @property
+    def dead_iter(self) -> int:
+        """Returns the stopping criteria: the number of iterations with no improvement, after
+        which the algorithm converges"""
+        return self._dead_iter
+
+    @dead_iter.setter
+    def dead_iter(self, value: int):
+        """
+        Sets the stopping criteria: the number of iterations with no improvement after which the
+        algorithm converges and stops.
+
+        Args:
+            value: Number of iterations with no improvement after which the algorithm converges.
+        """
+        self._dead_iter = value
+
+    @property
+    def best_cost_global(self) -> float:
+        """Returns the best individual cost found until the moment"""
+        return self._best_cost_global
+
+    @property
+    def best_ind_global(self):
+        """Returns the best individual instance found until the moment"""
+        return self._best_ind_global
+
+    @property
+    def generation(self):
+        """Returns the actual generation of solutions"""
+        return self._generation
+
+    @generation.setter
+    def generation(self, value: np.array):
+        """
+        Sets the generation to be evaluated. The user may want to initialize the algorithm from a
+        specific set of solutions.
+
+        Args:
+            value: set of solutions to be set as generation.
+        """
+        self._generation = value
```

Comment:
The only getter/setters should be for the parameters that are passed in the constructor so its configuration can be read and/or updated. Since the text seems to imply these are values that you can get while the minimize is running, if these are of interest to a user maybe the optimizer could have a user supplied callback that will be called at each iteration, if its provided, and get passed across values of interest such as these. That way the optimization process can be monitored. For instance like this https://github.com/Qiskit/qiskit-terra/blob/ed4e1a397d7faa38274f98c54ba5750043c32929/qiskit/algorithms/optimizers/qnspsa.py#L145

Reply:
Okay, I have removed the generation getter/setter as this might not be useful.

Reply:
Sorry, I had meant, by marking this whole block the getter/setter for `dead_iter`, and getters for `best_cost_global` and `best_ind_global`. I.e from above
>The only getter/setters should be for the parameters that are passed in the constructor

If you feel access to these are really need in some way then I had a suggestion above.

Reply:
Okay, I should have missunderstood this. I have just removed the getters/setters for dead_iter and the getters for best_cost and best_ind